gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/DagActionStoreChangeMonitor.java

```diff
@@ -146,15 +149,40 @@ protected void initializeMonitor() {
       throw new RuntimeException(String.format("Unable to retrieve dagActions from the dagActionStore while "
           + "initializing the %s", DagActionStoreChangeMonitor.class.getCanonicalName()), e);
     }
-    // TODO: make this multi-threaded to add parallelism
+    ThreadFactory threadFactory = getThreadFactoryWithUncaughtExceptionHandler();
+
+    final ExecutorService executorService = Executors.newFixedThreadPool(ConfigUtils.getInt(this.config, ConfigurationKeys.DAG_ACTION_STORE_MONITOR_EXECUTOR_THREADS, 5), threadFactory);
```

Comment:
hey, you can use `ExecutorUtils.newThreadFactory()` to get thread factory. it also has exception handler
https://github.com/apache/gobblin/blob/master/gobblin-utility/src/main/java/org/apache/gobblin/util/ExecutorsUtils.java#L76

Reply:
addressed